PWR/PWR_button/Core/Src/main.c

```diff
@@ -0,0 +1,182 @@
+#include "stm32l053xx.h"
+
+#define BUTTON_DEBOUNCE	50
+
+uint8_t flag_EXTI 		= 0,
+		user_bt_count	= 0;	//counter to debounce the button
+
+void MCU_Init(void);
+
+void EXTI4_15_IRQHandler (void)
+{
+	if(EXTI->PR & EXTI_PR_PIF10)	//if there's EXTI interrupt
+	{
+		flag_EXTI = 1;
+		EXTI->PR |= EXTI_PR_PIF10;	//clears the EXTI flag
+	}
+}
+
+int main(void)
+{
+	MCU_Init();
+
+	while(1)
+	{
+		if(flag_EXTI)
+		{
+			if(user_bt_count < 250)					//limits the counting to 250
+			{
+				user_bt_count++;					//counts up for debouncing
+			}
+
+			if((user_bt_count >= BUTTON_DEBOUNCE) && (GPIOA->IDR & GPIO_IDR_ID10_Msk))
+			{
+				GPIOA->ODR &= ~GPIO_ODR_OD7_Msk;	//turn the green led off
+				EXTI->PR |= EXTI_PR_PIF10;			//clears EXTI flag again before entering stop mode
+
+				SCB->SCR = SCB_SCR_SLEEPDEEP_Msk; 	//low-power mode = stop mode
+				PWR->CR &= ~PWR_CR_PDDS;			//making sure we're entering stop mode
+				PWR->CR |= PWR_CR_CWUF;				//clears WUF after 2 system clock cycles
+
+				PWR->CR |= 	PWR_CR_LPSDSR	|		//voltage regulator in low-power mode
+							PWR_CR_ULP		;		//ultra low power mode enable
+				//LPSDSR bit must be set before the LPRUN bit is set
+				PWR->CR |=	PWR_CR_LPRUN;			//voltage regulator in low-power mode
+
+				//entering stop mode procedure
+//				DBGMCU->CR |= DBGMCU_CR_DBG_STOP;	//this bit needs to be set if you're going to debug this code
+
+				__WFI();	//stop mode
+
+				while(!(GPIOA->IDR & GPIO_IDR_ID10_Msk));	//holds here to wake up from stop mode (doesn't enter in IRQ handler)
+
+				MCU_Init();
```

Comment:
As stop mode retains register values, we probably don't need to initialise everything again.